File: app/src/main/java/com/example/ahhosseini/rxjava2/network/CityService.java
package com.example.ahhosseini.rxjava2.network;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by ahHosseini on 2/12/18.
 */

public interface CityService {
    @GET("citiesJSON")
    Single<CityResponse> queryGeoname(
            @Query("north") double north,
            @Query("south") double south,
            @Query("east") double east,
            @Query("west") double west,
            @Query("lang") String lang);
}

File: app/src/main/java/com/example/ahhosseini/rxjava2/network/CityService.kt
package com.example.ahhosseini.rxjava2.network

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by ahHosseini on 2/12/18.
 */

interface CityService {
    @GET("citiesJSON")
    fun queryGeoname(
            @Query("north") north: Double,
            @Query("south") south: Double,
            @Query("east") east: Double,
            @Query("west") west: Double,
            @Query("lang") lang: String): Single<CityResponse>
}
